tests/test_azure.py

```diff
@@ -502,3 +502,13 @@ def test_blobserviceclient_no_custom_domain(self):
                 "https://foo_name.blob.core.windows.net",
                 credential={"account_name": "foo_name", "account_key": "foo_key"},
             )
+
+    def test_missing_file_handling(self):
+        client_mock = mock.MagicMock()
+        client_mock.exists.side_effect = [True, False]
+        self.storage._client.get_blob_client.return_value = client_mock
+
+        file_that_exists = self.storage._open('file-that-exists')
```

Comment:
Can you change this to use `open()` as that is the public API. Also to use a `with` context handler.

Reply:
I've just pushed commits to make these changes